reposilite-backend/src/main/kotlin/com/reposilite/maven/Repository.kt

```diff
@@ -56,11 +57,18 @@ class Repository internal constructor(
         val sha1 = location.resolveSibling(location.getSimpleName() + ".sha1")
         val sha256 = location.resolveSibling(location.getSimpleName() + ".sha256")
         val sha512 = location.resolveSibling(location.getSimpleName() + ".sha512")
-
-        return storageProvider.putFile(md5, DigestUtils.md5Hex(bytes).byteInputStream())
-            .flatMap { storageProvider.putFile(sha1, DigestUtils.sha1Hex(bytes).byteInputStream()) }
-            .flatMap { storageProvider.putFile(sha256, DigestUtils.sha256Hex(bytes).byteInputStream()) }
-            .flatMap { storageProvider.putFile(sha512, DigestUtils.sha512Hex(bytes).byteInputStream()) }
+        val temporaryFile = File.createTempFile("reposilite-", "-file-checksum")
+
+        bytes.use { input ->
+            temporaryFile.outputStream().use { output ->
+                input.copyTo(output)
+            }
+        }
+
+        return storageProvider.putFile(md5, DigestUtils.md5Hex(temporaryFile.inputStream()).byteInputStream())
+            .flatMap { storageProvider.putFile(sha1, DigestUtils.sha1Hex(temporaryFile.inputStream()).byteInputStream()) }
+            .flatMap { storageProvider.putFile(sha256, DigestUtils.sha256Hex(temporaryFile.inputStream()).byteInputStream()) }
+            .flatMap { storageProvider.putFile(sha512, DigestUtils.sha512Hex(temporaryFile.inputStream()).byteInputStream()) }
```

Comment:
Good question. I just assumed since it was getting an InputStream prior that it was being handled. After looking through the source, i do not believe that it does: